redis/src/main/java/org/hibernate/ogm/datastore/redis/dialect/model/impl/DocumentBasedAssociation.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.redis.dialect.model.impl;
+
+import java.util.List;
+
+import org.hibernate.ogm.datastore.redis.dialect.value.Association;
+import org.hibernate.ogm.datastore.redis.dialect.value.StructuredValue;
+
+
+/**
+ * A {@link RedisAssociation} backed by an {@link Association}.
+ *
+ * @author Gunnar Morling
+ */
+class DocumentBasedAssociation extends RedisAssociation {
```

Comment:
The classes in this package look very similar to what we have for CouchDB. Do you think this could be extracted into shared classes living in core?